cdap-api/src/main/java/co/cask/cdap/api/data/DatasetContext.java

```diff
@@ -59,4 +57,43 @@
   <T extends Dataset> T getDataset(String name, Map<String, String> arguments)
     throws DatasetInstantiationException;
 
+  /**
+   * Calling this means that the dataset is not used by the caller any more,
+   * and the DatasetContext is free to dismiss or reuse it. The dataset must
+   * have been acquired through this context using {@link #getDataset(String)}
+   * or {@link #getDataset(String, Map)}. It is up to the implementation of the
+   * context whether this dataset is discarded, or whether it is reused as the
+   * result for subsequent {@link #getDataset} calls with the same arguments.
+   * This may be influenced by resource constraints, expiration policy, or
+   * advanced configuration.
+   * <p>
+   * Note: In the current implementation, this is equivalent to {@link #discardDataset(Dataset)}.
```

Comment:
I mean as part of the API contract, should the user even care about it? And when we modify the implementation, we need to remember to go to this class and modify the javadoc, which seems counter-intuitive since an interface contract shouldn't be affected by the implementation.